### PR TITLE
Prepare for 0.24 release

### PR DESCRIPTION
**Pull request by Misty-W**
## Description

Update changelog and version for 0.24 release

---

### License

- [x] I license this contribution under the terms of the GNU GPL, version 3 and grant Unitary Fund the right to provide additional permissions as described in section 7 of the GNU GPL, version 3.

Before opening the PR, please ensure you have completed the following where appropriate.
- [x] I added unit tests for new code.
- [x] I used [type hints](https://www.python.org/dev/peps/pep-0484/) in function signatures.
- [x] I used [Google-style](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods) docstrings for functions.
- [x] I [updated the documentation](../blob/master/docs/CONTRIBUTING_DOCS.md) where relevant.
- [x] Added myself / the copyright holder to the AUTHORS file

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,35 @@
 # Changelog
 
-## Version 0.24.0 (In development)
+## Version 0.24.0
+
+Highlights of this release include refactoring of parts of the PEC module, improvements to the Calibration data and settings structures,
+completion of the REM section of the user guide, and the publishing of a Mitiq tutorial first presented as a lab exercise at the SQMS/GGI 2022 Summer School on Quantum Simulation of Field Theories.
+Special thanks to UF Ambassadors Amir Ebrahimi and Purva Thakre for their contributions to this release!
+
```
Comment (andreamari) [inline CHANGELOG.md:8]: It's probably worth adding something like this.

```suggestion

**Breaking changes:** The class `NoisyOperation` is deprecated and removed from Mitiq. Moreover the initialization arguments of the `OperationRepresentation` class changed. Please check the associated API-docs and the [PEC section](https://mitiq.readthedocs.io/en/stable/guide/pec.html) of the docs for more details.
```

**Review by andreamari — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,36 @@
...
+**Breaking changes:** The class `NoisyOperation` is deprecated and removed from Mitiq. Moreover the initialization arguments of the `OperationRepresentation` class changed. Please check the associated API-docs and the [PEC section](https://mitiq.readthedocs.io/en/stable/guide/pec.html) of the docs for more details.
+- Refactoring of NoisyOperation, NoisyBasis, OperationRepresentation (#1712) [@andreamari]
```
Comment (Misty-W) [inline CHANGELOG.md:10]:
```suggestion
- Update pennylane-qiskit requirement from ~=0.28.0 to ~=0.29.0 (#1733) [@dependabot[bot]]
- Update pennylane requirement from ~=0.28.0 to ~=0.29.0 (#1732) [@dependabot[bot]]
- Refactoring of NoisyOperation, NoisyBasis, OperationRepresentation (#1712) [@andreamari]
```

**Comment by andreamari** [comment]
Hi @Misty-W, is there anything blocking this?

**Comment by Misty-W** [comment]
Nothing blocking- just merged :)

**Pull request merged**